apps/meteor/app/lib/server/functions/getUserSingleOwnedRooms.ts

```diff
@@ -10,23 +10,25 @@ export const getUserSingleOwnedRooms = function (subscribedRooms: SubscribedRoom
 	const roomsThatWillBeRemoved = subscribedRooms.filter(({ shouldBeRemoved }) => shouldBeRemoved).map(({ rid }: { rid: string }) => rid);
 
 	const roomIds = roomsThatWillBeRemoved.concat(roomsThatWillChangeOwner);
-	const rooms = Rooms.findByIds(roomIds, { fields: { _id: 1, name: 1, fname: 1 } });
+	const rooms = Rooms.findByIds(roomIds, { projection: { _id: 1, name: 1, fname: 1 } });
 
 	const result = {
 		shouldBeRemoved: [] as string[],
 		shouldChangeOwner: [] as string[],
 	};
 
-	rooms.forEach((room: IRoom) => {
-		const name = room.fname || room.name;
-		if (roomsThatWillBeRemoved.includes(room._id)) {
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			result.shouldBeRemoved.push(name!);
-		} else {
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			result.shouldChangeOwner.push(name!);
-		}
-	});
+	Promise.await(
```

Comment:
Why a promise.await here?

Reply:
because its how cursor works , its going to run for every record and then resolve